script/Deployer.sol

```diff
@@ -50,13 +53,17 @@ contract Deployer is Script {
         poolManager =
             new PoolManager(address(escrow), liquidityPoolFactory, restrictionManagerFactory, trancheTokenFactory);
 
-        LiquidityPoolFactory(liquidityPoolFactory).rely(address(poolManager));
-        TrancheTokenFactory(trancheTokenFactory).rely(address(poolManager));
-        RestrictionManagerFactory(restrictionManagerFactory).rely(address(poolManager));
+        AuthLike(liquidityPoolFactory).rely(address(poolManager));
+        AuthLike(trancheTokenFactory).rely(address(poolManager));
+        AuthLike(restrictionManagerFactory).rely(address(poolManager));
+
+        AuthLike(liquidityPoolFactory).rely(address(root));
+        AuthLike(trancheTokenFactory).rely(address(root));
+        AuthLike(restrictionManagerFactory).rely(address(root));
 
-        LiquidityPoolFactory(liquidityPoolFactory).rely(address(root));
-        TrancheTokenFactory(trancheTokenFactory).rely(address(root));
-        RestrictionManagerFactory(restrictionManagerFactory).rely(address(root));
+        AuthLike(liquidityPoolFactory).deny(deployer);
```

Comment:
Why not put these in `removeDeployerAccess`?

Reply:
the factories are not global vars

Reply:
Let's make them global vars then. There's two reasons I think it should be in `removeDeployerAccess`:
1. This method is not called in the permissionless deploy script. This is quite helpful for testing, we can deploy while keeping the deployer access.
2. It is easier to audit that all `deny`s are done.